p2p/swarmhandlers.go

```diff
@@ -552,6 +570,9 @@ func (s *swarmImpl) sendMessageSendError(cr net.ConnectionError) {
 
 // TODO : move peer store management out of swarm
 func (s *swarmImpl) onGetPeerRequest(gpr getPeerRequest) {
+	defer s.peerMapMutex.Unlock()
```

Comment:
We shouldn't use defer (apparently)

Reply:
grokbase.com/t/gg/golang-nuts/158zz5p42w/go-nuts-defer-performance
https://github.com/golang/go/issues/14939